prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -319,4 +355,94 @@ object CustomFHIRFunctions {
         }
         return if (type != null) mutableListOf(StringType(type)) else mutableListOf()
     }
+
+    /**
+     * Get the LOINC Code from the LIVD table based on the device id, equipment model id, test kit name id, or the
+     * element model name
+     * @return a list with one value denoting the LOINC Code, or an empty list
+     */
+    fun lookupLivdTableLoincCodes(
+        focus: MutableList<Base>,
+        parameters: MutableList<MutableList<Base>>?,
+        metadata: Metadata = Metadata.getInstance()
+    ): MutableList<Base> {
+        val lookupTable = metadata.findLookupTable(name = "LIVD-SARS-CoV-2")
+        val observation = focus.first()
```

Comment:
You need more checks here. Focus can be empty.  Also, what happens if there is more than one?